### ISSUE TITLE
[Feature Request] Need better messaging than "No sources found. Is OBS running?" when OBS integration may not be configured

### ISSUE THREAD

**Issue by wrenNdoubt** [crowbartools/Firebot#2308]
**Help Request**
Brand new Firebot user here. Firebot is linked to my Streamer and Bot accounts. The URL has been pasted into a browser source in OBS, and is working if I test a command by playing a video. But when I try to add the effect "Toggle OBS Source Visibility," under sources, it says, "No sources found. Is OBS running?"

OBS is, in fact, running. 

Clicking "refresh source data" does nothing. I'm using the most up-to-date version of OBS. I don't know how to fix this. I googled for help but couldn't find it. Any troubleshooting tips would be appreciated.
![Screenshot_4](https://github.com/crowbartools/Firebot/assets/154454701/b371adc9-fa62-4635-a19c-2249f87d0937)
![Screenshot_5](https://github.com/crowbartools/Firebot/assets/154454701/a9eb930e-9806-4442-8365-9c7b820600cc)

**Comment by CKY-** [comment]
> **Help Request** Brand new Firebot user here. Firebot is linked to my Streamer and Bot accounts. The URL has been pasted into a browser source in OBS, and is working if I test a command by playing a video. But when I try to add the effect "Toggle OBS Source Visibility," under sources, it says, "No sources found. Is OBS running?"
> 
> OBS is, in fact, running.
> 
> Clicking "refresh source data" does nothing. I'm using the most up-to-date version of OBS. I don't know how to fix this. I googled for help but couldn't find it. Any troubleshooting tips would be appreciated. ![Screenshot_4](https://private-user-images.githubusercontent.com/154454701/292956968-b371adc9-fa62-4635-a19c-2249f87d0937.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NTIxODUsIm5iZiI6MTcwMzY1MTg4NSwicGF0aCI6Ii8xNTQ0NTQ3MDEvMjkyOTU2OTY4LWIzNzFhZGM5LWZhNjItNDYzNS1hMTljLTIyNDlmODdkMDkzNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyN1QwNDM4MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNGU4YzlhMDcwNTFmMjVhODBiODJjYWZhYjAyMDUzMjdmYWJlMDAyNDNmNzI5YmJmNDZlOTBkMmU5ZGRjZjVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.h2YdXWMV1y_3j94wSx5jsGKavTVIgCHETyPvihsBlN8) ![Screenshot_5](https://private-user-images.githubusercontent.com/154454701/292956970-a9eb930e-9806-4442-8365-9c7b820600cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM2NTIxODUsIm5iZiI6MTcwMzY1MTg4NSwicGF0aCI6Ii8xNTQ0NTQ3MDEvMjkyOTU2OTcwLWE5ZWI5MzBlLTk4MDYtNDQ0Mi04MzY1LTljN2I4MjA2MDBjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyN1QwNDM4MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOTY2ZmQxMmQzZGM1MjEwZTllN2E5NmI5OWE1NDQ1NzlmZThkMjVmY2VjZmRmYTUzMzJjZGNhMGY0OTM4N2UyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.z9SG2O8UtNMuJ-ja8mjt83Sqf8__f5vrzw9gSYGKLcU)

here are some common questions that might help narrow this down abit  

did you enable the server in OBS? 
     on OBS Tools>obs-websockets-server
             is it enabled 
             did you set a password and enable require authentication 
             what is the port?

did you in integrations set the same password and port from OBS in the obs config of Firebot
     on firebot Left hand side Settings>integrations> OBS config
     are the ports the same?
     is this the same PC?
          yes 
               is the address set to 127.0.0.1 or localhost
           no 
                did you use the address provided by OBS it is a "best guess" and not always correct

**Comment by CKY-** [comment]
I think where the confusion lies is in the wording of whats presented, 
"No sources found. Is OBS running?"

Maybe we could do better checking to see if the integration is configured and show a different message based on that.
but in fact the question should be is the websocket server enabled or a more generic Err:Could not connect to OBS WebSocket server

**Comment by wrenNdoubt** [comment]
Thank you for these answers. I did not see anything about websocket servers in the tutorials or google searches I did, RIP.

So I enabled the websocket server in OBS.

Port: 4455
Authentication enabled yes
Generated new password

Firebot, port is the same, entered the new password. IP says localhost since it's the same computer.

For a few seconds it showed Firebot connected as a websocket on OBS, but when I tried to set up the Toggle OBS Source Visibility, it still has the same message and can't find any sources. It doesn't seem to stick as a websocket.
